File: itsm-backend/ent/schema/problem.go
package schema

import (
	"time"

	"entgo.io/ent"
	"entgo.io/ent/schema/field"
)

// Problem holds the schema definition for the Problem entity.
type Problem struct {
	ent.Schema
}

// Fields of the Problem.
func (Problem) Fields() []ent.Field {
	return []ent.Field{
		field.String("title").
			Comment("问题标题").
			NotEmpty(),
		field.Text("description").
			Comment("问题描述").
			Optional(),
		field.String("status").
			Comment("状态").
			Default("open"),
		field.String("priority").
			Comment("优先级").
			Default("medium"),
		field.String("category").
			Comment("问题分类").
			Optional(),
		field.Text("root_cause").
			Comment("根本原因").
			Optional(),
		field.Text("impact").
			Comment("影响范围").
			Optional(),
		field.Int("assignee_id").
			Comment("处理人ID").
			Optional(),
		field.Int("created_by").
			Comment("创建人ID").
			Positive(),
		field.Int("tenant_id").
			Comment("租户ID").
			Positive(),
		field.Time("created_at").
			Comment("创建时间").
			Default(time.Now),
		field.Time("updated_at").
			Comment("更新时间").
			Default(time.Now).
			UpdateDefault(time.Now),
		field.Time("resolved_at").
			Comment("解决时间").
			Optional().
			Nillable(),
		field.Time("closed_at").
			Comment("关闭时间").
			Optional().
			Nillable(),
		field.Time("deleted_at").
			Comment("删除时间").
			Optional().
			Nillable(),
	}
}

// Edges of the Problem.
func (Problem) Edges() []ent.Edge {
	return nil
}

File: itsm-frontend/src/types/biz/problem.ts
/**
 * 问题管理类型定义
 */

import { ProblemPriority, ProblemStatus } from '@/constants/problem';

// 问题实体接口
export interface Problem {
  id: number;
  title: string;
  description: string;
  status: ProblemStatus;
  priority: ProblemPriority;
  category: string;
  rootCause?: string;
  impact?: string;
  assigneeId?: number;
  createdBy: number;
  tenantId: number;
  createdAt: string;
  updatedAt: string;
}

// 创建问题请求
export interface CreateProblemRequest {
  title: string;
  description: string;
  priority: ProblemPriority;
  category: string;
  rootCause?: string;
  impact?: string;
}

// 更新问题请求
export interface UpdateProblemRequest {
  title?: string;
  description?: string;
  status?: ProblemStatus;
  priority?: ProblemPriority;
  category?: string;
  rootCause?: string;
  impact?: string;
  assigneeId?: number;
}

// 列表查询参数
export interface ProblemQuery {
  page?: number;
  pageSize?: number;
  status?: string;
  priority?: string;
  category?: string;
  keyword?: string;
}

// 列表响应
export interface ProblemListResponse {
  problems: Problem[];
  total: number;
  page: number;
  pageSize: number;
}

// 统计响应
export interface ProblemStats {
  total: number;
  open: number;
  inProgress: number;
  resolved: number;
  closed: number;
  highPriority: number;
}
